superset/result_set.py

```diff
@@ -95,9 +96,10 @@ def __init__(
             # generate numpy structured array dtype
             numpy_dtype = [(column_name, "object") for column_name in column_names]
 
-        # put data in a structured array so we can efficiently access each column.
-        # cast `data` as list due to MySQL (others?) wrapping results with a tuple.
-        array = np.array(list(data), dtype=numpy_dtype)
+        # only do expensive recasting if datatype is not standard list of tuples
+        if data and not isinstance(data, list) and not isinstance(data[0], tuple):
+            data = [tuple(row) for row in data]
+        array = np.array(data, dtype=numpy_dtype)
```

Comment:
@robdiciuccio I changed the logic so that this expensive operation only takes place when needed.